src/VisualStudio/Core/Def/Implementation/UnusedReferences/ProjectAssets/ProjectAssetsReader.cs

```diff
@@ -82,61 +86,62 @@ internal static class ProjectAssetsReader
                 return null;
             }
 
-            return BuildReference(projectAssets, target, referenceName, referenceInfo.TreatAsUsed);
+            return BuildReference(projectAssets, referenceName, referenceInfo.TreatAsUsed);
         }
 
         private static ReferenceInfo? BuildReference(
             ProjectAssetsFile projectAssets,
-            Dictionary<string, ProjectAssetsTargetLibrary> target,
-            string dependency,
+            string referenceName,
             bool treatAsUsed)
         {
-            var key = target.Keys.FirstOrDefault(library => library.Split('/')[0] == dependency);
-            if (key is null)
+            var dependencyNames = new HashSet<string>();
+            var compilationAssemblies = ImmutableArray.CreateBuilder<string>();
+            var referenceType = ReferenceType.Unknown;
+
+            var packagesPath = projectAssets.Project?.Restore?.PackagesPath ?? string.Empty;
+
+            foreach (var target in projectAssets.Targets!.Values)
```

Comment:
```suggestion
            RoslynDebug.AssertNotNull(projectAssets.Targets);
            RoslynDebug.AssertNotNull(projectAssets.Libraries);
            
            foreach (var target in projectAssets.Targets.Values)
```

Reply:
Adding these makes it so you don't have to add `!` in a lot of places, makes the expectation clearer